Places: New design for folder icons

newhoa opened the pull request:
Combines the newer design of the `system-file-manager` app icon with the color palette of the current `folder` icons. Also adjusts the aspect ratio to make it slightly wider, so it is not quite as square as `system-file-manager` but not quite as wide as the current `folder` icons.

Closes #1305

---

This is what has been done as of the initial PR:

- Basic folder at all sizes
- XDG User Directories at 48px and 64px
- folder-open (drag-accept) at 48px

---

My initial thoughts/questions:

~1) I am thinking the 16px folder should be slightly less rounded?~ (went ahead and adjusted this)
2) I just added `folder-open` at 48px, feedback for that one would be appreciated before I start on any other sizes.
4) Is the aspect ratio of `folder` okay? (personally, I am liking it a lot the more I use them so I hope that can stick).

Any feedback would be appreciated!

Thanks!

danirabbit reviewed (commented):
These are so good! I especially like how much more open the folder-open is here. It looks like it has a lot more dimension to it. Great job

Proportions of everything looks good, and I noticed these fit the baseline better compared to other icons which is really great.

Honestly these are really lovely and they do look way more modern which is great. I just had a few little nitpicks, so I opened a branch I figured that would be easier than trying to type it out: https://github.com/newhoa/icons/pull/1

newhoa commented:
Okay, I think these are ready for review!

Went ahead and adjusted the shadows, gradients, and highlights for all sizes as suggested by @danirabbit [here](https://github.com/newhoa/icons/pull/1). Added `folder-open` remaining sizes, did the XDG emblemed folders, etc.

Let me know if anything needs adjusting!

Thanks!

teamcons commented:
> Okay, I think these are ready for review!
> 
> Went ahead and adjusted the shadows, gradients, and highlights for all sizes as suggested by @danirabbit [here](https://github.com/newhoa/icons/pull/1). Added `folder-open` remaining sizes, did the XDG emblemed folders, etc.
> 
> Let me know if anything needs adjusting!
> 
> Thanks!

I am using it on our daily computer
A nitpick would be the Files manager icon, too high

![Bildschirmfoto von 2024-11-09 12 24 32](https://github.com/user-attachments/assets/3b996c45-8009-4aa4-8349-0d82373904b5)

Various parts of the interface have icons with the old pointy folder. But it may be just my eyes

![Bildschirmfoto von 2024-11-09 12 29 17](https://github.com/user-attachments/assets/6b77e9f9-8e06-4226-b94a-c1987c60494f)

Each commit you do makes your hairs shinier and softer

danirabbit reviewed (changes requested):
@newhoa these look awesome! Do you think we could also do actions:
- [ ] document-save-as
- [ ] folder-copy
- [ ] folder-move
- [ ] folder-new

It looks like the inside of the folder highlight gradient stop needs to be adjusted at 24px in folder/folder-open

I think we discussed before having a different aspect ratio between folder icons in Files and the Files app icon itself is okay.

newhoa commented:
> @newhoa these look awesome! Do you think we could also do actions:
> 
>     * [ ]  document-save-as
> 
>     * [ ]  folder-copy
> 
>     * [ ]  folder-move
> 
>     * [ ]  folder-new
> 

Should I make another PR for the action icons since they're a different category?

Either way, I'm wondering if the `folder-open` at 24px should be a simple rectangle (like it is at 16px, and like `folder-open` is currently at 24px) instead of using a v-shape. Or maybe a more rectangular version for the action icons only? The v-shape seems to make it loose its sharpness, and I think the sharper look is nicer for action icons at these small sizes at least.

![current](https://github.com/user-attachments/assets/4202ad83-e6af-4cbc-8a3d-11fda7a26054)

![prop1](https://github.com/user-attachments/assets/e442b918-1a33-4ad0-af25-b7d2592f2509)

Anyway, let me know if you want these added in this PR or to start another PR for the action icons.

teamcons commented:
Id go with B. 

Very personal but the arrow in front of the folder, like it points to something on the ground in front of it
feels very jarring for the purpose of it. It doesnt represent "in" the folder, just "oops miss the folder go in front"

It feels like it requires rework, outside of the round folder thing

danirabbit commented:
@newhoa 
> Should I make another PR for the action icons since they're a different category?

I think it makes sense to merge them all at once :woman_shrugging: 

> Personally I like B here

I'm happy with that! 

Thanks so much for working on these :)

newhoa commented:
@danirabbit Updated the action icons.

There were a few `Merge branch 'main' into places-rounded-folder` commits, I wasn't totally sure what to do (or how to resolve it) before this new commit. I just did `git pull` then added the "Update action icons" commit. But then the git log looks odd, has commits for this branch, then the commits from main, and now another commit for this branch. Wasn't sure if I did that right, if not let me know please!

As for

>It looks like the inside of the folder highlight gradient stop needs to be adjusted at 24px in folder/folder-open

I'm not sure exactly where you mean. The left side highlight only has 3 stops? That's the only difference I can tell from the highlights at the other sizes. But the 24px `system-file-manager` icon also only had 3 stops so it is unchanged from there. Anyway, just let me know if that is what it is and I can change it to the 4 stop gradient (or if it's something else just let me know that too). Thanks!

danirabbit commented:
Don't worry about commit history, we use squash merge anyways :)

danirabbit reviewed (approved):
Gonna go ahead and merge these. They look fantastic. Nice work!
